### PR TITLE
add support for parsing HTML documents

### LINKED ISSUE #115: Add support for `<html>` tags in templates
I've [opened up a PR](https://github.com/open-wc/open-wc/pull/2282/files) in open-wc's eslint-plugin-lit-a11y package to implement an eslint rule that checks to see if the `<html>'s `lang` attribute receives a compliant value. (W3 validate against this rule, see [here](https://www.w3.org/International/techniques/authoring-html.en?open=language&open=textprocessing#textprocessing) and [here](https://www.w3.org/International/techniques/authoring-html.en?open=language&open=langvalues#langvalues))

Unfortunately, since we're using `parse5 's`parseFragment` function to generate the AST, the resulting object strips out the elements like `<HTML>`, `<body>`, and `<head>`, which is not good since we want to run validation against the `<html>` element.

The way I've gotten around this is by extending the `template-analyzer` to check if an element is an `<html>` element via the following function 

```javascript
function isRootElement(node) {
  // You can find a breakdown of this regex pattern here: https://regex101.com/r/lKJKFv/1
  return /<html([\s\S]*)<\/html>/.test(node);
}
```

Then inside of the template-analyzer's constructor we make the following modification:

```javascript
  const html = this.templateExpressionToHtml(node);
    const options = {
      treeAdapter,
      sourceCodeLocationInfo: true,
      onParseError: err => {
        this.errors.push(err);
      },
    });
    };

    if (isRootElement(html)) {
      this._ast = parse5.parse(html, options);
    } else {
      this._ast = parse5.parseFragment(html, options);
    }
```

I created a PR to implement this [here](https://github.com/open-wc/open-wc/pull/2282/) but this causes open-wc's template-analyzer to become out sync with the one written in `eslint-plugin-lit`. 

@thepassle left [a comment in the PR](https://github.com/open-wc/open-wc/pull/2282/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#r705067783) about possibly importing your template-analyzer instead of open-wc rolling their own.

> We've recently had some discussions to see what the deltas between our template analyzer, and the original template analyzer are, and see if we can just import the original template analyzer instead. If we make changes here, it'll be harder to move back to the original template analyzer.

If you think it appropriate, I'd be happy to create a PR to add the `isRootElement` check, to enable your package, and other consumers of the `template-analyzer` to add lint rules to the `<html>` element.

### PR DESCRIPTION
**Pull request by 43081j**
Fixes #115 

This is another utility one really, adding the ability for template analyzer to parse HTML documents.

It always parses as a fragment at the min, which means something like this:

```html
<html><body>Foo</body></html>
```

would be parsed as `Foo` essentially.

by adding this switch in the constructor, we'll parse it as a document instead, maintaining the html/body tags.

cc @stramel 

Part im not sure of is if the condition is strong enough, i.e. checking for `includes('<html')`. seems incredibly unlikely anyone can ever have a `<html-something-else` tag so maybe its ok?

Also looks like prettier being bumped now decided to re-format URLs

### CODE REVIEW

**Comment by 43081j** [comment]
one thing to consider maybe is people doing stuff like this:

```ts
html`<div attr=${5<html}></div>`;
```

i suppose that'd make it fall over... but without an absurd amount of pre-parsing im not sure how we can catch such things

**Comment by andrico1234** [comment]
@43081j , if you like i can checkout this branch, link it to the lit-a11y eslint repo, and run the test suite? 

Will that provide any assurance that everything still works as expected?

**Comment by 43081j** [comment]
Sure that would be useful. I think I just need to add some more tests here too for the edge cases, will probably have time tomorrow

File: src/template-analyzer.ts [hunk 1]
```diff
@@ -94,7 +94,11 @@ export class TemplateAnalyzer {
       }
     };
 
-    this._ast = parse5.parseFragment(this.source, opts);
+    if (this.source.includes('<html')) {
```
Comment (stramel) [inline src/template-analyzer.ts:97]: I'm okay with starting out with a basic check but this doesn't account for `<HTML` which is still valid and I know there are some devs write it like that still.

Comment (stramel) [reply src/template-analyzer.ts:97]: I'm not too worried about the commented case. That seems unlikely.

**Review by stramel — APPROVED** [review]

File: src/template-analyzer.ts (hunk 1 above)
Comment (43081j) [reply src/template-analyzer.ts:97]: ah fair point, so i probably should use a regex after all. was trying to avoid it since this'd be faster, but it won't be if i have to lowercase it all :joy: will update before we merge
